ide/static/js/canvas.js

```diff
@@ -272,13 +274,18 @@ class Canvas extends React.Component {
     if (!this.clickOrDraggedLayer) {
       this.clickOrDraggedLayer = 1;
     }
-    const layerId = event.el.id;
-    const layer = this.props.net[layerId];
-    layer.state.left = `${event.pos['0']}px`;
-    layer.state.top = `${event.pos['1']}px`;
-    this.props.modifyLayer(layer, layerId);
-    let net = this.props.net
-    this.checkCutting(net);
+    this.lastEvent = event;
+  }
+  mouseUpEvent(event, layerId){
+    if(this.lastEvent) {
+      const layer = this.props.net[layerId];
+      layer.state.left = `${this.lastEvent.pos['0']}px`;
+      layer.state.top = `${this.lastEvent.pos['1']}px`;
+      this.props.modifyLayer(layer, layerId);
```

Comment:
@ZeroZeroJedenJeden rather than modifying the layer details for every mouse event can't we have a regular interval let's say 5 events and then make update. Because the current implementation still doesn't help much with the issue.

Reply:
Because modifyLayer call does quite a lot of computation which is not needed to be changed for every mouse event.

Reply:
mouseUpEvent activate when you release left mouse buttom.

Reply:
So, "mouseUpEvent" function is executed only one time for all dragging one element.

Reply:
Okay, let me test on my system.